htmresearch/algorithms/faulty_temporal_memory.py

```diff
@@ -23,26 +23,109 @@
 Faulty Temporal Memory implementation in Python.
 """
 import numpy
-from collections import defaultdict
-from htmresearch.algorithms.temporal_memory_phases import TemporalMemory
+from nupic.research.temporal_memory import  TemporalMemory
+from nupic.research.connections import CellData, Connections
+
+
+
+class FaultyCellData(CellData):
+  """ CellData subclass that adds a `destroyed` property that evaluates to True
+  when a cell is "killed"
+  """
+
+
+  __slots__ = CellData.__slots__ + ["__destroyed"]
+
+
+  def __init__(self):
+    super(FaultyCellData, self).__init__()
+    self.__destroyed = False
+
+
+  @property
+  def destroyed(self):
+    return self.__destroyed
+
+
+  def destroy(self):
+    self.__destroyed = True # Destruction is permanent!
+
+
+
+class FaultyConnections(Connections):
```

Comment:
Is the FaultyConnections needed? It seems it'd be easiest to just keep a set of `self.deadCells` on the `FaultyTemporalMemory`.

Then just never grow a segment from a dead cell, and destroy all segments on a cell when destroying the cell. All this logic can live in the `FaultyTemporalMemory`.